ding_dibs.email_reports.inc

```diff
@@ -0,0 +1,292 @@
+<?php
+/**
+ * @file
+ * E-mail reports functionality.
+ */
+
+
+/**
+ * Form settings mailing reports.
+ *
+ * @return
+ *   Save results form settings.
+ */
+function ding_dibs_setting_emails($form, &$form_state) {
+  drupal_set_title(t('Ding DIBS setting emails'));
+  drupal_add_css(drupal_get_path('module', 'ding_dibs') . '/css/ding_dibs.css');
+  $hour = range(0, 23);
+  $options_hour = array_map(generate_zerro, $hour);
+  $minute = range(0, 59);
+  $options_minute = array_map(generate_zerro, $minute);
+  $text = '<label for="edit-ding-dibs-start-cron">';
+  $text .= t('Set the daily hour to run this cron');
+  $text .= '<span title="' . t('This field is required.');
+  $text .= '" class="form-required">*</span></label>';
```

Comment:
There is something going on here with html generated by hand and inserted into `#prefix` and ´#suffix`. I am not sure why but I am guessing it has something to do with inlining form elements.

1. Without code comments it is hard to understand what is going on.
2. I do not think this is the best approach for achieving this. Have you considered using `fieldset` or nested form elements? NB. I have not tested whether the approach below works but the approach chosen here doesn not seem very elegant.

```php
$form['ding_dibs_start_cron'] = array(
  '#prefix' => '<div class="container-inline">',
  '#suffix' => '</div>',
  '#tree' => TRUE,
);
$form['ding_dibs_start_cron']['hour'] = array(
  '#type' => 'select',
  '#options' => $options_hour,
  // more here
);
$form['ding_dibs_start_cron']['minute'] = array(
  '#type' => 'select',
  '#options' => $options_minute,
);
```